File: app/src/main/java/com/juniperphoton/myerlistandroid/widget/CircleView.java
package com.juniperphoton.myerlistandroid.widget;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.view.View;

public class CircleView extends View {

    private int mColor;
    private Paint mPaint;

    public CircleView(Context context, AttributeSet attrs) {
        super(context, attrs);
        mPaint = new Paint();
        mPaint.setColor(Color.WHITE);
    }

    public void setColor(int color) {
        mColor = color;
        if (mPaint == null) {
            mPaint = new Paint();
        }
        mPaint.setColor(mColor);
        invalidate();
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        canvas.drawCircle(getWidth() / 2, getHeight() / 2, getWidth() / 2, mPaint);
    }
}

File: app/src/main/java/com/juniperphoton/myerlist/widget/CircleView.kt
package com.juniperphoton.myerlist.widget

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.util.AttributeSet
import android.view.View

open class CircleView(context: Context, attrs: AttributeSet?) : View(context, attrs) {
    private val paint: Paint = Paint()

    open var color: Int = Color.WHITE
        set(value) {
            field = value
            paint.color = this.color
            invalidate()
        }

    init {
        paint.color = Color.WHITE
        paint.isDither = true
        paint.isAntiAlias = true
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        canvas.drawCircle((width / 2).toFloat(), (height / 2).toFloat(), (width / 2).toFloat(), paint)
    }
}
